### PR TITLE
Ruby 3.0 support + GitHub Actions (MSP-Greg)

### PR DESCRIPTION
**Pull request by johnnyshields**
This includes:
- commits from #59
- adds Ruby 3.0 support.
- removes travis.yml entirely
- drops support for Ruby <= 2.1 (users on older Ruby can use an earlier version of this gem)
- cleanup gempsec and add Ruby >= 2.2 condition to gemspec
- removes unneeded magic comments `coding: utf-8`

### CODE REVIEW

File: lib/unf_ext/version.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module UNF
   class Normalizer
-    VERSION = "0.0.7.7"
+    VERSION = "0.0.7.8"
```
Comment (knu) [inline lib/unf_ext/version.rb:3]: Could you remove this change?

File: unf_ext.gemspec [hunk 2]
```diff
@@ -1,42 +1,30 @@
...
-  gem.add_development_dependency("rdoc", ["> 2.4.2"])
-  gem.add_development_dependency("bundler", [">= 1.2"])
-  gem.add_development_dependency("rake-compiler", [">= 0.7.9"])
-  gem.add_development_dependency("rake-compiler-dock", [">= 1.0.1"])
+  gem.add_development_dependency('rake', ['>= 0.9.2.2'])
+  gem.add_development_dependency('test-unit')
+  gem.add_development_dependency('rdoc', ['> 2.4.2'])
+  gem.add_development_dependency('bundler', ['>= 1.2'])
+  gem.add_development_dependency('rake-compiler', ['>= 0.7.9'])
+  gem.add_development_dependency('rake-compiler-dock', ['>= 1.0.1'])
```
Comment (knu) [inline unf_ext.gemspec:29]: Please keep the diff to a minimum.

**Review by knu — CHANGES_REQUESTED** [review]
Thanks!

I'd like style changes separated from functional changes because it's hard to review when the two types of changes mixed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,10 @@
-## 0.0.7.7 (2019-03-19)
+## 0.0.7.8 (2021-08-29)
```
[the remark this answers is not in the document]
Comment (johnnyshields) [reply CHANGELOG.md]: Done.

File: lib/unf_ext/version.rb (hunk 1 above)
Comment (johnnyshields) [reply lib/unf_ext/version.rb]: Done > reverted

File: unf_ext.gemspec (hunk 2 above)
Comment (johnnyshields) [reply unf_ext.gemspec]: Done > Diff minimized.

**Comment by johnnyshields** [comment]
@knu I think it's difficult to support Ruby < 2.2 on Github Actions, that's the reason for dropping it.

Since the published versions of this gem support older Ruby, and Ruby 2.1 was EOL in 2017, it's probably not an issue.

**Review by knu — COMMENTED** [review]
Thanks!  I think I can squash and split the commits into meaningful sets of hunks, but do you intend to do that yourself?

**Comment by johnnyshields** [comment]
@knu I've squashed the commits and included @MSP-Greg 's original commit.

**Review by knu — APPROVED** [review]
Thanks!  I'm merging this and will prepare for a new release.

**Pull request merged**